### ISSUE TITLE
t.w.static.File lies in content-length when byterange falls off the end of the file

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3814]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)| @mwhudson reported|
|-|-|
|Trac ID|trac#3814|
|Type|defect|
|Created|2009-05-04 08:48:04Z|

```/foo``` is a 30 byte static resource.  That makes it hard to serve 990 bytes out of it, which is what twisted.web claims it's going to do:

```
mwh@grond:Twisted$ telnet localhost 8081
Trying 127.0.0.1...
Connected to localhost.
Escape character is '^]'.
GET /foo HTTP/1.1
Host: localhost
Range: bytes=10-1000
Connection: close
 
HTTP/1.1 206 Partial Content
Content-Length: 990
Accept-Ranges: bytes
Content-Range: bytes 10-999/31
Server: TwistedWeb/8.2.0+r26805
Last-Modified: Mon, 04 May 2009 08:40:47 GMT
Connection: close
Date: Mon, 04 May 2009 08:42:26 GMT
Content-Type: text/html

01234567890123456789
^]

telnet> close
```

I guess I'll fix this at the same time as I fix [#3574](https://github.com/twisted/twisted/issues/3574).

<details><summary>Searchable metadata</summary>

```
trac-id__3814 3814
type__defect defect
reporter__mwh mwh
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1241426884000000 1241426884000000
changetime__1246312861000000 1246312861000000
version__None None
owner__mwh mwh

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_1"></a>|@mwhudson commented|
|-|-|

(In [26906]) test and fix for when the range falls off the end of the resource.

refs [#3814](https://github.com/twisted/twisted/issues/3814)

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_2"></a>|@mwhudson set status to `closed`|
|-|-|

(In [27029]) Merge t.w.static.File-multiple-ranges-3574-2

Author: mwhudson
Reviewer: therve
Fixes: [#3574](https://github.com/twisted/twisted/issues/3574)
Fixes: [#3814](https://github.com/twisted/twisted/issues/3814)

Support multiple byteranges in the Range: header in requests for
static resources.

In the process fix a bug with Content-Length being wrong when the
request range only partially overlaps with the resource.
